backend/src/rss-feed/rss-feed.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import sources from './rss-feed-sources';
+import { RssFeed } from './rss-feed.entity';
+import { Repository } from 'typeorm';
+const Parser = require('rss-parser');
+const parser = new Parser();
+
+
+@Injectable()
+export class RssFeedService {
+
+    constructor(@InjectRepository(RssFeed)
+    private readonly rssFeedRepository: Repository<RssFeed>) {
+
+    }
+
+    async onModuleInit() {
+        await this.fetchAndSaveRssFeeds();
+    }
+
+    async fetchAndSaveRssFeeds() {
+        const news = await parser.parseURL(sources.news);
+        await this.rssFeedRepository.save(news.items);
+        const stories = await parser.parseURL(sources.stories);
+        await this.rssFeedRepository.save(stories.items);
+    }
+
+    async findAll() {
```

Comment:
isnt this a recursive call? What does this return.

Reply:
This returns all the rss feeds. I will add the return type here, if thats what you are asking?.

Reply:
Yes It was wrong :) I will update